tuxtape-server/src/main.rs

```diff
@@ -1,12 +1,25 @@
 use clap::Parser;
 use cli::Cli;
 use color_eyre::{Result, eyre::eyre};
+use grpc::server;
+use tonic::transport::{Identity, ServerTlsConfig};
 
 mod cli;
+mod grpc;
 
 #[tokio::main]
 async fn main() -> Result<()> {
     let args = Cli::parse();
+    let server_addr = args.addr;
+    let server_tls_config = match args.tls {
+        true => {
+            let cert = std::fs::read_to_string(&args.tls_cert_path)?;
+            let key = std::fs::read_to_string(&args.tls_key_path)?;
+            let identity = Identity::from_pem(cert, key);
+            Some(ServerTlsConfig::new().identity(identity))
+        }
+        false => None,
+    };
```

Comment:
`.then` will convert a boolean to an option.
```suggestion
    let server_tls_config = args.tls.then(|| {
            let cert = std::fs::read_to_string(&args.tls_cert_path)?;
            let key = std::fs::read_to_string(&args.tls_key_path)?;
            let identity = Identity::from_pem(cert, key);
            ServerTlsConfig::new().identity(identity)
        });
```